packages/ui/components/ui/tx/view/value.tsx

```diff
@@ -36,15 +37,34 @@ export const ValueViewComponent = ({ view }: ValueViewProps) => {
   if (view.valueView.case === 'knownAssetId') {
     const value = view.valueView.value;
     const amount = value.amount ?? new Amount();
-    const display_denom = value.metadata?.display ?? '';
+    const displayDenom = getDisplayDenomFromView(view);
     const exponent = value.metadata ? getDisplayDenomExponent(value.metadata) : 0;
 
     return (
       <div className='flex font-mono'>
-        {fromBaseUnitAmount(amount, exponent).toFormat()} {display_denom}
+        {fromBaseUnitAmount(amount, exponent).toFormat()} {showDenom && displayDenom}
       </div>
     );
   }
 
   return <></>;
 };
+
+export const getDisplayDenomFromView = (view: ValueView) => {
+  if (view.valueView.case === 'unknownAssetId') {
+    if (!view.valueView.value.assetId) throw new Error('no asset id for unknown denom');
+    return bech32AssetId(view.valueView.value.assetId);
+  }
+
+  if (view.valueView.case === 'knownAssetId') {
+    const displayDenom = view.valueView.value.metadata?.display;
+    if (displayDenom) return displayDenom;
+
+    const assetId = view.valueView.value.metadata?.penumbraAssetId;
+    if (assetId) return bech32AssetId(assetId);
+
+    return 'unknown';
+  }
+
+  throw new Error(`unexpected case ${view.valueView.case}`);
```

Comment:
Maybe you're intending this, but keep in mind that `view.valueView.case` can be `undefined` if there isn't any value to display. Should this really throw in that case?

Reply:
Yeah, that's fair. Going to return `unknown`.